lib/queue.js

```diff
@@ -1024,7 +1024,13 @@ Queue.prototype.processJob = function(job) {
   }
 
   function handleFailed(err) {
-    var error = err.cause || err; //Handle explicit rejection
+    var error = err;
```

Comment:
I prefer to just make the code simpler and just pass the err argument directly to the user in the failed event.

Reply:
@manast That would actually cause the user's error to *not* be returned in the following case:
```js
queue.process(function(job, callback) {
  jobDone(new Error());
});
```

Instead, a `Bluebird.OperationalError` would be passed to the `failed` handler. This arguably leaks implementation details to the caller and would be a change in API since it's not currently returned. It would also break this test case that explicitly checks for this:
https://github.com/OptimalBits/bull/blob/cf2c6883fc2d861397bd59df7186b730800655e0/test/test_queue.js#L1208-L1232

My PR was an attempt to have the best of both worlds: unwrap Bull's internal error but don't unwrap too much.

Reply:
ok, I get it.